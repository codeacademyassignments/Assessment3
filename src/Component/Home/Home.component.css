.mainBlock{
  margin: 0% 10%;
}

.rootSection{
  background-color: grey;
}