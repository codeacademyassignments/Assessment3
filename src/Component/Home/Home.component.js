import React, { Component } from 'react';
import axios from 'axios';
import PropTypes from 'prop-types';
import Header from '../Header/Header.component';
import Section from '../Section/Section.component';
import testData from '../../TestData/TestData';
import './Home.component.css';


// import ReactDOM from 'react-dom';
// import RawData from '../../RawData/RawData';

class Home extends Component {
  componentDidMount() {
    const { onDataStoreComplete, onDataFetched, fetchState } = this.props;
    if (!fetchState) {
      const allBookGroupByAuthor = axios.get('http://localhost:8080/allBooksWithRating');
      return allBookGroupByAuthor.then((response) => {
        onDataFetched(response.data);
        onDataStoreComplete();
      });
    }
    return {};
  }

  render() {
    const { blogs } = this.props;
    // console.log('here..', blogs);
    if (Object.keys(blogs).length === 0) return (<></>);
    return (
      <section className="rootSection">
        <Header />
        <section className="mainBlock">
          <Section cards={blogs['J K Rowling']} author="J K Rowling" />
          <Section cards={blogs['Sidney Sheldon']} author="Sidney Sheldon" />
        </section>
      </section>
    );
  }


  // componentDidUpdate(prevProps) {
  //   console.log('component updated', prevProps);
  // }
}

Home.propTypes = {
  blogs: PropTypes.arrayOf(
    PropTypes.shape({
      date: PropTypes.string.isRequired,
      readingTime: PropTypes.string.isRequired,
      title: PropTypes.string.isRequired,
      description: PropTypes.string.isRequired,
      liked: PropTypes.bool.isRequired,
      claps: PropTypes.number.isRequired,
      image: PropTypes.string,
    }),
  ),
  onDataStoreComplete: PropTypes.func.isRequired,
  onDataFetched: PropTypes.func.isRequired,
  fetchState: PropTypes.bool.isRequired,
};

Home.defaultProps = {
  blogs: [],
};


export default Home;
