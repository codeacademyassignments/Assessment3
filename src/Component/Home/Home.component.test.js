import React from 'react';
import { shallow } from 'enzyme';
import axios from 'axios';
import renderer from 'react-test-renderer';
import Home from './Home.component';
import testData from '../../TestData/TestData';

describe('Home Component', () => {
  it('snapshot should match', () => {
    const tree = renderer.create(<Home onDataFetched={jest.fn()} />);
    expect(tree).toMatchSnapshot();
  });
});

describe('componentDidMount', () => {
  let mockAxios;
  let mockFetchComplete;
  beforeAll(() => {
    mockAxios = jest.fn().mockImplementation(() => new Promise(((resolve) => { resolve({ data: testData }); })));
    axios.get = mockAxios;
    mockFetchComplete = jest.fn();
  });
  afterAll(() => {
    jest.resetModules();
  });
  it('should call all mock functions with proper arguements', async (done) => {
    const wrapper = shallow(<Home onDataFetched={mockFetchComplete} />);
    await wrapper.instance().componentDidMount();
    expect(mockAxios).toHaveBeenCalledWith('http://localhost:8080/allBooksWithRating');

    expect(mockFetchComplete).toHaveBeenCalledWith(testData);
    done();
  });
  it('should return empty object when fetchState is true', async () => {
    const wrapper = shallow(<Home books={testData} onDataFetched={mockFetchComplete} />);
    expect(await wrapper.instance().componentDidMount()).toEqual({});
  });
});
