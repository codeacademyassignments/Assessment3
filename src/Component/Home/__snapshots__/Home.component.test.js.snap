// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Component snapshot should match 1`] = `null`;
